### PR TITLE
Refactor navigation and state handling

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Implement a new navigation controller which also manages the view models. This is mainly a preparation for adding more subscreens and allowing to open more than one ORT result at once later on. Please see the commit messages for details.

### CODE REVIEW

File: src/main/kotlin/navigation/NavController.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        if (initialScreen != null) {
+            navigate(initialScreen, unique = false)
+        }
+    }
+
+    /**
+     * Navigate to the provided [screen]. If [unique] is true and there is already an equal screen in the backstack,
+     * this screen will be moved to the top of the backstack instead of creating a new one.
+     */
+    fun navigate(screen: Screen<*>, unique: Boolean = true) {
```
Comment (sschuberth) [inline src/main/kotlin/navigation/NavController.kt:36]: Is "reuse" a better term than "unique", as it maybe better expresses what happens if a view is already preset?

File: src/main/kotlin/Main.kt [hunk 2]
```diff
@@ -9,10 +9,10 @@ import androidx.compose.ui.window.WindowState
 import androidx.compose.ui.window.singleWindowApplication
 
 import org.ossreviewtoolkit.workbench.ui.App
-import org.ossreviewtoolkit.workbench.ui.AppState
+import org.ossreviewtoolkit.workbench.ui.WorkbenchController
```
Comment (sschuberth) [inline src/main/kotlin/Main.kt:13]: Commit message nit: Capitalize "Rename" in the title.

File: src/main/kotlin/navigation/BackstackEntry.kt [hunk 3]
```diff
@@ -9,3 +9,8 @@ class BackstackEntry(
     val screen: Screen<*>,
     val viewModel: ViewModel
 )
+
```
Comment (sschuberth) [inline src/main/kotlin/navigation/BackstackEntry.kt:12]: Commit message nits:
- "in *the* future"
- "navigation *for* parameterized subscreen*s*"

File: src/main/kotlin/lifecycle/MoleculeViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ */
+abstract class MoleculeViewModel<EVENT, MODEL> : ViewModel() {
+    /**
+     * The [CoroutineScope] used to [launch][launchMolecule] molecule.
+     */
+    private val moleculeScope = CoroutineScope(scope.coroutineContext + MainUIDispatcher)
+
+    /**
+     * A flow of UI events to handle. The capacity is large enough to handle simultaneous UI events, but small enough to
+     * surface issues if they get backed up for some reason.
```
[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply src/main/kotlin/lifecycle/MoleculeViewModel.kt:27]: I took the wording from the official example:
https://github.com/cashapp/molecule/blob/1e5809a45885cb3b5a2c4373c63cc3de760bf7aa/sample-viewmodel/src/main/java/com/example/molecule/viewmodel/MoleculeViewModel.kt#L32-L34

File: src/main/kotlin/navigation/NavController.kt (hunk 1 above)
Comment (mnonnenmacher) [reply src/main/kotlin/navigation/NavController.kt]: I went with `launchSingleTop` to stay close to the Android navigation:
https://developer.android.com/jetpack/compose/navigation#nav-to-composable

**Review by sschuberth — APPROVED** [review]

**Pull request merged**